src/app/rpmostree-builtin-status.cxx

```diff
@@ -273,6 +273,48 @@ get_last_auto_update_run (GDBusConnection   *connection,
   return TRUE;
 }
 
+/* Get the StatusText property of `update_driver_sd_unit`, assuming it is a service unit. */
+static gboolean
+get_update_driver_sd_status (RPMOSTreeSysroot *sysroot_proxy,
+                             const char       *update_driver_sd_unit,
+                             const char      **update_driver_status,
+                             GCancellable     *cancellable,
+                             GError          **error)
+{
+  GDBusConnection *connection =
+    g_dbus_proxy_get_connection (G_DBUS_PROXY (sysroot_proxy));
+
+  /* Query systemd for update driver's systemd unit's object path. */
+  g_autoptr(GVariant) update_driver_objpath_tuple = 
+    g_dbus_connection_call_sync (connection, "org.freedesktop.systemd1", "/org/freedesktop/systemd1",
+                                 "org.freedesktop.systemd1.Manager", "LoadUnit",
+                                 g_variant_new ("(s)", update_driver_sd_unit), G_VARIANT_TYPE_TUPLE,
+                                 G_DBUS_CALL_FLAGS_NONE, -1, cancellable, error);
+  if (!update_driver_objpath_tuple || g_variant_n_children (update_driver_objpath_tuple) < 1)
+    return FALSE;
```

Comment:
If `update_driver_objpath_tuple` is not NULL, then no error happened, so we shouldn't just `return FALSE` here. Probably something like:

```c
if (!update_driver_objpath_tuple)
  return FALSE;
else if (g_variant_n_children (update_driver_objpath_tuple) < 1)
  return glnx_throw (error, "LoadUnit(%s) returned empty result", update_driver_sd_unit)
```

Reply:
ah right. thanks for pointing that out!